src/coprocessor/dag/rpn_expr/types/expr_builder.rs

```diff
@@ -119,12 +137,23 @@ fn get_eval_type(tree_node: &Expr) -> Result<EvalType> {
 }
 
 #[inline]
-fn handle_node_column_ref(tree_node: Expr, rpn_nodes: &mut Vec<RpnExpressionNode>) -> Result<()> {
+fn handle_node_column_ref(
+    tree_node: Expr,
+    rpn_nodes: &mut Vec<RpnExpressionNode>,
+    max_columns: usize,
+) -> Result<()> {
     let offset = number::decode_i64(&mut tree_node.get_val()).map_err(|_| {
         Error::Other(box_err!(
             "Unable to decode column reference offset from the request"
         ))
     })? as usize;
+    if offset >= max_columns {
```

Comment:
any test cover it?

Reply:
This is a simple `if a >= b` check. I think it is not very necessary to verify that this line is correct or not.